client/src/components/Search/SearchBox.js

```diff
@@ -9,134 +9,141 @@ import ResizeAware from 'components/ResizeAware/ResizeAware';
 import PropTypes from 'prop-types';
 import TagPropType from '../Tag/TagPropType';
 
+const defaultBorders = {
+  top: false,
+  right: false,
+  bottom: true,
+  left: true,
+};
+
 /**
  * Displays a search box and a few buttons related buttons.
  */
-class SearchBox extends Component {
-  constructor(props) {
-    super(props);
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-    this.handleFocus = this.handleFocus.bind(this);
-    this.handleBlur = this.handleBlur.bind(this);
-    this.renderInput = this.renderInput.bind(this);
-    this.renderFilterButton = this.renderFilterButton.bind(this);
-    this.renderEnterHint = this.renderEnterHint.bind(this);
-    this.renderHideButton = this.renderHideButton.bind(this);
-    this.componentDidUpdate = this.componentDidUpdate.bind(this);
-    this.onResize = this.onResize.bind(this);
-    this.setWidth = this.setWidth.bind(this);
-    this.renderTags = this.renderTags.bind(this);
-    this.getComponentWidth = this.getComponentWidth.bind(this);
-    this.calculateSpaceForTags = this.calculateSpaceForTags.bind(this);
-    this.calculateInputLeftPadding = this.calculateInputLeftPadding.bind(this);
-    this.calculateInputRightPadding = this.calculateInputRightPadding.bind(this);
-    this.onTagListResize = this.onTagListResize.bind(this);
-    this.focusOnLastTag = this.focusOnLastTag.bind(this);
-    this.focusOnInput = this.focusOnInput.bind(this);
-
-    this.state = {
-      hasFocus: false,
-      width: window.innerWidth - 180 - 55,
-      tagWidth: 0
-    };
-  }
-
-  componentDidUpdate() {
-    const width = this.getComponentWidth();
-    this.setWidth(width);
-  }
+const SearchBox = ({
+  onSearch,
+  onToggleFilter,
+  onHideFilter,
+  onChange,
+  onHide,
+  onTagDelete,
+  onTagClick,
+  placeholder = i18n._t('Admin.SEARCH', 'Search'),
+  expanded,
+  formId,
+  id,
+  searchText,
+  hideable,
+  showFilters,
+  name,
+  dirty,
+  clearable,
+  tagData = [],
+  borders = {
+    top: false,
+    right: false,
+    bottom: true,
+    left: true,
+  },
```

Comment:
```suggestion
  borders = defaultBorders,
```
This was the reason for moving that outside the component.

Reply:
Updated